misk-aws/src/main/kotlin/misk/jobqueue/sqs/SqsJobConsumer.kt

```diff
@@ -63,12 +67,22 @@ internal class SqsJobConsumer @Inject internal constructor(
   }
 
   internal inner class QueueReceiver(
-    private val queueName: QueueName,
+    queueNames: List<QueueName>,
     private val handler: JobHandler
   ) {
-    private val queue = queues[queueName]
+    private val prioritizedQueues = queueNames.map { queues[it] }
 
     fun runOnce(): Status {
+      prioritizedQueues.forEach { resolvedQueue ->
+        val status = runOnce(resolvedQueue)
+        if (status != Status.NO_WORK) {
```

Comment:
how should we handle failures from a queue? still skip processing lower prioritized queues?

Reply:
similar question, what about exceptions?

Reply:
on failures / exceptions we retry the same queue again. i believe this is the correct behaviour. the bad items should be pushed to the DLQ and we get the next batch.

Reply:
but you potentially starve the lower priority queues. is that ok?

Reply:
yes, i think that's correct, since you shouldn't get stuck processing the same message -- it should get pushed to the DLQ. there's no way to know if the top priority queue is full of error-causing messages, or if it's just one, so i think you need to keep processing them.

Reply:
> there's no way to know if the top priority queue is full of error-causing messages, or if it's just one, so i think you need to keep processing them.

that's not true. we could do something, like inspect the current error rate.

Reply:
We can know how many errors we've processed, but not how many are still to come. Might be a reasonable heuristic there, but I'd worry that it's overcomplicating things.